tests/test_base_device.py

```diff
@@ -1114,3 +1174,171 @@ def qiskit_circuit(x):
+        qiskit_dev = QiskitDevice2(wires=3, backend=backend, shots=30000)
+
+        @qml.qnode(qiskit_dev)
+        def qiskit_circuit():
+            qml.RX(np.pi / 3, 0)
+            qml.RZ(np.pi / 3, 0)
+            return observable()
+
+        dev = qml.device("default.qubit", wires=3, shots=30000)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(np.pi / 3, 0)
+            qml.RZ(np.pi / 3, 0)
+            return observable()
+
+        qiskit_res = qiskit_circuit()
+        res = circuit()
+
+        equals = lambda a, b: abs(a - b) < 300  # tolerance of 1% of # of shots
+
+        for i in range(len(res)):  # pylint: disable=consider-using-enumerate
+            for key in set(res[i].keys()) & set(qiskit_res[i].keys()):
+                assert equals(res[i][key], qiskit_res[i][key])
+
+    @pytest.mark.parametrize(
+        "observable",
+        [
+            lambda: [qml.sample(qml.X(0) + qml.Y(0)), qml.sample(qml.X(0))],
+            lambda: [
+                qml.sample(qml.X(0) @ qml.Z(1) + 0.5 * qml.Y(1) + qml.Z(0)),
+                qml.sample(0.5 * qml.Y(1)),
+            ],
+            lambda: [
+                qml.sample(qml.X(0)),
+                qml.sample(qml.Y(1)),
+                qml.sample(0.5 * qml.Y(1)),
+                qml.sample(qml.Z(0) @ qml.Z(1)),
+                qml.sample(qml.X(0) @ qml.Z(1) + 0.5 * qml.Y(1) + qml.Z(0)),
+                qml.sample(
+                    qml.ops.LinearCombination(
+                        [0.35, 0.46], [qml.X(0) @ qml.Z(1), qml.Z(0) @ qml.X(2)]
+                    )
+                ),
+                qml.sample(
+                    qml.ops.LinearCombination(
+                        [1.0, 2.0, 3.0], [qml.X(0), qml.X(1), qml.Z(0)], grouping_type="qwc"
+                    )
+                ),
+            ],
+            lambda: [
+                qml.sample(
+                    qml.Hamiltonian([0.35, 0.46], [qml.X(0) @ qml.Z(1), qml.Z(0) @ qml.Y(2)])
+                )
+            ],
+        ],
+    )
+    def test_observables_that_need_split_non_commuting_samples(self, observable):
+        """Tests that observables that have non-commuting measurents are processed
+        correctly when executed by the Sampler via sample()"""
+        qiskit_dev = QiskitDevice2(wires=3, backend=backend, shots=30000)
+
+        @qml.qnode(qiskit_dev)
+        def qiskit_circuit():
+            qml.RX(np.pi / 3, 0)
+            qml.RZ(np.pi / 3, 0)
+            return observable()
+
+        dev = qml.device("default.qubit", wires=3, shots=30000)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(np.pi / 3, 0)
+            qml.RZ(np.pi / 3, 0)
+            return observable()
+
+        qiskit_res = qiskit_circuit()
+        res = circuit()
+
+        for i in range(len(res)):  # pylint: disable=consider-using-enumerate
+            assert abs(sum(qiskit_res[i]) / 30000 - sum(res[i]) / 30000) < 0.05
```

Comment:
What does `ith` index give here for the results?
Can we do something like `np.abs(np.sum(qiskit_res - res, axis=0)) / 30000 < 0.05`

Reply:
`res` or `qiskit_res` are an array of arrays, where the inner arrays are the results of the `qml.sample`. 
Could do `assert np.all(np.abs(np.sum(qiskit_res, axis=1) / 30000 - np.sum(res, axis=1) / 30000) < 0.05)` instead.